src/nextcloud/talk/recording/Server.py

```diff
@@ -164,9 +171,40 @@ def _getAddressWithoutPort(self, address):
 
         return address
 
+class ProtectedMetrics:
+    """
+    Middleware to serve the metrics only if the remote address is allowed to
+    access them.
+    """
+
+    # pylint: disable=redefined-outer-name
+    def __init__(self, config):
+        self.config = config
+        self.metrics = make_wsgi_app()
+
+    def __call__(self, environment, startResponse):
+        """
+        Returns the metrics if the remote address is allowed to access them, or
+        an error 404 if not.
+        """
+
+        remoteAddress = environment['REMOTE_ADDR']
+        remoteAddress = ip_address(remoteAddress)
+
+        if not isAddressInNetworks(remoteAddress, config.getStatsAllowedIps()):
+            startResponse('403 FORBIDDEN', [])
```

Comment:
Either the documentation is wrong, or you're returning the wrong error here?

Reply:
Guess who changed his mind after writing the documentation and forgot to update it :facepalm: Nice catch :-)